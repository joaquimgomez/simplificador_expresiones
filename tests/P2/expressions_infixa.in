INFIXA ( 2 + 5 ) * 3 + 1 -> INFIXA
INFIXA 2 + 5 * 3 + 1 -> INFIXA
INFIXA ( a and ( b or ( not ( T and a ) ) ) ) -> INFIXA
INFIXA ( a and ( b or ( not T and a ) ) ) -> INFIXA
INFIXA ( a and ( b or not T and a ) ) -> INFIXA
INFIXA ( a and b or not T and a ) -> INFIXA
INFIXA a and b or not T and a -> INFIXA
INFIXA ( ( F or a ) or b ) and ( a or ( b and b ) ) -> INFIXA
INFIXA ( F or a or b ) and ( a or b and b ) -> INFIXA
INFIXA F or a or b and a or b and b -> INFIXA
INFIXA ( 12 - 9 ) / ( ( 23 + 5 ) * -2 ) -> INFIXA
INFIXA ( 12 - 9 ) / ( 23 + 5 * -2 ) -> INFIXA
INFIXA ( 12 - 9 ) / 23 + 5 * -2 -> INFIXA
INFIXA 12 - 9 / ( 23 + 5 * -2 ) -> INFIXA
INFIXA 12 - 9 / 23 + 5 * -2 -> INFIXA
INFIXA ( ( x + 3 ) + ( 2 * y ) ) - ( ( 2 * y ) + ( 3 + x ) ) -> INFIXA
INFIXA ( x + 3 + 2 * y ) - ( 2 * y + 3 + x ) -> INFIXA
INFIXA x + 3 + 2 * y - 2 * y + 3 + x -> INFIXA
INFIXA ( x + 12 ) + ( 12 + x ) -> INFIXA
INFIXA ( x + 12 ) + 0 -> INFIXA
INFIXA 0 + ( 12 + x ) -> INFIXA
INFIXA ( x + 12 ) - ( 12 + x ) -> INFIXA
INFIXA ( x + 12 ) - 0 -> INFIXA
INFIXA ( x + 12 ) * ( 12 + x ) -> INFIXA
INFIXA ( x + 12 ) * 0 -> INFIXA
INFIXA 0 * ( 12 + x ) -> INFIXA
INFIXA ( x + 12 ) * 1 -> INFIXA
INFIXA 1 * ( 12 + x ) -> INFIXA
INFIXA ( x + 12 ) / ( 12 + x ) -> INFIXA
INFIXA ( x + 12 ) / 1 -> INFIXA
INFIXA 0 / ( 12 + x ) -> INFIXA
INFIXA ( ( T == c ) or d ) and e -> INFIXA
INFIXA ( T == ( c or d ) ) and e -> INFIXA
INFIXA T == ( c or d and e ) -> INFIXA
INFIXA T == c or d and e -> INFIXA
INFIXA aa and ( bb or ( cc == T ) ) -> INFIXA
INFIXA aa and ( ( bb or cc ) == T ) -> INFIXA
INFIXA ( aa and bb or cc ) == T -> INFIXA
INFIXA aa and bb or cc == T -> INFIXA
INFIXA ( ( ( a / b ) * c ) - d ) + e -> INFIXA
INFIXA ( ( a / b * c ) - d ) + e -> INFIXA
INFIXA ( a / b * c - d ) + e -> INFIXA
INFIXA a / b * c - d + e -> INFIXA
INFIXA a / b * c - ( d + e ) -> INFIXA
INFIXA a / b * ( c - ( d + e ) ) -> INFIXA
INFIXA a / ( b * ( c - ( d + e ) ) ) -> INFIXA
INFIXA a1 + ( b2 - ( c3 * ( d4 / e5 ) ) ) -> INFIXA
INFIXA a1 + ( b2 - ( c3 * d4 / e5 ) ) -> INFIXA
INFIXA a1 + ( b2 - c3 * d4 / e5 ) -> INFIXA
INFIXA a1 + b2 - c3 * d4 / e5 -> INFIXA
INFIXA a1 + ( b2 - c3 ) * d4 / e5 -> INFIXA
INFIXA ( a1 + ( b2 - c3 ) ) * d4 / e5 -> INFIXA
INFIXA ( a1 + ( b2 - c3 ) ) * ( d4 / e5 ) -> INFIXA
INFIXA ( c and d ) and ( a and b ) -> INFIXA
INFIXA c and d and a and b -> INFIXA
INFIXA not ( c and d ) and not ( a and b ) -> INFIXA
INFIXA not not ( c and d ) and not not ( a and b ) -> INFIXA
INFIXA not c and not d and not a and not b -> INFIXA
INFIXA not not c and not not d and not not a and not not b -> INFIXA
INFIXA ( 2 + 5 ) * 3 + 1 -> PREFIXA
INFIXA 2 + 5 * 3 + 1 -> PREFIXA
INFIXA ( a and ( b or ( not ( T and a ) ) ) ) -> PREFIXA
INFIXA ( a and ( b or ( not T and a ) ) ) -> PREFIXA
INFIXA ( a and ( b or not T and a ) ) -> PREFIXA
INFIXA ( a and b or not T and a ) -> PREFIXA
INFIXA a and b or not T and a -> PREFIXA
INFIXA ( ( F or a ) or b ) and ( a or ( b and b ) ) -> PREFIXA
INFIXA ( F or a or b ) and ( a or b and b ) -> PREFIXA
INFIXA F or a or b and a or b and b -> PREFIXA
INFIXA ( 12 - 9 ) / ( ( 23 + 5 ) * -2 ) -> PREFIXA
INFIXA ( 12 - 9 ) / ( 23 + 5 * -2 ) -> PREFIXA
INFIXA ( 12 - 9 ) / 23 + 5 * -2 -> PREFIXA
INFIXA 12 - 9 / ( 23 + 5 * -2 ) -> PREFIXA
INFIXA 12 - 9 / 23 + 5 * -2 -> PREFIXA
INFIXA ( ( x + 3 ) + ( 2 * y ) ) - ( ( 2 * y ) + ( 3 + x ) ) -> PREFIXA
INFIXA ( x + 3 + 2 * y ) - ( 2 * y + 3 + x ) -> PREFIXA
INFIXA x + 3 + 2 * y - 2 * y + 3 + x -> PREFIXA
INFIXA ( x + 12 ) + ( 12 + x ) -> PREFIXA
INFIXA ( x + 12 ) + 0 -> PREFIXA
INFIXA 0 + ( 12 + x ) -> PREFIXA
INFIXA ( x + 12 ) - ( 12 + x ) -> PREFIXA
INFIXA ( x + 12 ) - 0 -> PREFIXA
INFIXA ( x + 12 ) * ( 12 + x ) -> PREFIXA
INFIXA ( x + 12 ) * 0 -> PREFIXA
INFIXA 0 * ( 12 + x ) -> PREFIXA
INFIXA ( x + 12 ) * 1 -> PREFIXA
INFIXA 1 * ( 12 + x ) -> PREFIXA
INFIXA ( x + 12 ) / ( 12 + x ) -> PREFIXA
INFIXA ( x + 12 ) / 1 -> PREFIXA
INFIXA 0 / ( 12 + x ) -> PREFIXA
INFIXA ( ( T == c ) or d ) and e -> PREFIXA
INFIXA ( T == ( c or d ) ) and e -> PREFIXA
INFIXA T == ( c or d and e ) -> PREFIXA
INFIXA T == c or d and e -> PREFIXA
INFIXA aa and ( bb or ( cc == T ) ) -> PREFIXA
INFIXA aa and ( ( bb or cc ) == T ) -> PREFIXA
INFIXA ( aa and bb or cc ) == T -> PREFIXA
INFIXA aa and bb or cc == T -> PREFIXA
INFIXA ( ( ( a / b ) * c ) - d ) + e -> PREFIXA
INFIXA ( ( a / b * c ) - d ) + e -> PREFIXA
INFIXA ( a / b * c - d ) + e -> PREFIXA
INFIXA a / b * c - d + e -> PREFIXA
INFIXA a / b * c - ( d + e ) -> PREFIXA
INFIXA a / b * ( c - ( d + e ) ) -> PREFIXA
INFIXA a / ( b * ( c - ( d + e ) ) ) -> PREFIXA
INFIXA a1 + ( b2 - ( c3 * ( d4 / e5 ) ) ) -> PREFIXA
INFIXA a1 + ( b2 - ( c3 * d4 / e5 ) ) -> PREFIXA
INFIXA a1 + ( b2 - c3 * d4 / e5 ) -> PREFIXA
INFIXA a1 + b2 - c3 * d4 / e5 -> PREFIXA
INFIXA a1 + ( b2 - c3 ) * d4 / e5 -> PREFIXA
INFIXA ( a1 + ( b2 - c3 ) ) * d4 / e5 -> PREFIXA
INFIXA ( a1 + ( b2 - c3 ) ) * ( d4 / e5 ) -> PREFIXA
INFIXA ( c and d ) and ( a and b ) -> PREFIXA
INFIXA c and d and a and b -> PREFIXA
INFIXA not ( c and d ) and not ( a and b ) -> PREFIXA
INFIXA not not ( c and d ) and not not ( a and b ) -> PREFIXA
INFIXA not c and not d and not a and not b -> PREFIXA
INFIXA not not c and not not d and not not a and not not b -> PREFIXA
INFIXA ( 2 + 5 ) * 3 + 1 -> POSTFIXA
INFIXA 2 + 5 * 3 + 1 -> POSTFIXA
INFIXA ( a and ( b or ( not ( T and a ) ) ) ) -> POSTFIXA
INFIXA ( a and ( b or ( not T and a ) ) ) -> POSTFIXA
INFIXA ( a and ( b or not T and a ) ) -> POSTFIXA
INFIXA ( a and b or not T and a ) -> POSTFIXA
INFIXA a and b or not T and a -> POSTFIXA
INFIXA ( ( F or a ) or b ) and ( a or ( b and b ) ) -> POSTFIXA
INFIXA ( F or a or b ) and ( a or b and b ) -> POSTFIXA
INFIXA F or a or b and a or b and b -> POSTFIXA
INFIXA ( 12 - 9 ) / ( ( 23 + 5 ) * -2 ) -> POSTFIXA
INFIXA ( 12 - 9 ) / ( 23 + 5 * -2 ) -> POSTFIXA
INFIXA ( 12 - 9 ) / 23 + 5 * -2 -> POSTFIXA
INFIXA 12 - 9 / ( 23 + 5 * -2 ) -> POSTFIXA
INFIXA 12 - 9 / 23 + 5 * -2 -> POSTFIXA
INFIXA ( ( x + 3 ) + ( 2 * y ) ) - ( ( 2 * y ) + ( 3 + x ) ) -> POSTFIXA
INFIXA ( x + 3 + 2 * y ) - ( 2 * y + 3 + x ) -> POSTFIXA
INFIXA x + 3 + 2 * y - 2 * y + 3 + x -> POSTFIXA
INFIXA ( x + 12 ) + ( 12 + x ) -> POSTFIXA
INFIXA ( x + 12 ) + 0 -> POSTFIXA
INFIXA 0 + ( 12 + x ) -> POSTFIXA
INFIXA ( x + 12 ) - ( 12 + x ) -> POSTFIXA
INFIXA ( x + 12 ) - 0 -> POSTFIXA
INFIXA ( x + 12 ) * ( 12 + x ) -> POSTFIXA
INFIXA ( x + 12 ) * 0 -> POSTFIXA
INFIXA 0 * ( 12 + x ) -> POSTFIXA
INFIXA ( x + 12 ) * 1 -> POSTFIXA
INFIXA 1 * ( 12 + x ) -> POSTFIXA
INFIXA ( x + 12 ) / ( 12 + x ) -> POSTFIXA
INFIXA ( x + 12 ) / 1 -> POSTFIXA
INFIXA 0 / ( 12 + x ) -> POSTFIXA
INFIXA ( ( T == c ) or d ) and e -> POSTFIXA
INFIXA ( T == ( c or d ) ) and e -> POSTFIXA
INFIXA T == ( c or d and e ) -> POSTFIXA
INFIXA T == c or d and e -> POSTFIXA
INFIXA aa and ( bb or ( cc == T ) ) -> POSTFIXA
INFIXA aa and ( ( bb or cc ) == T ) -> POSTFIXA
INFIXA ( aa and bb or cc ) == T -> POSTFIXA
INFIXA aa and bb or cc == T -> POSTFIXA
INFIXA ( ( ( a / b ) * c ) - d ) + e -> POSTFIXA
INFIXA ( ( a / b * c ) - d ) + e -> POSTFIXA
INFIXA ( a / b * c - d ) + e -> POSTFIXA
INFIXA a / b * c - d + e -> POSTFIXA
INFIXA a / b * c - ( d + e ) -> POSTFIXA
INFIXA a / b * ( c - ( d + e ) ) -> POSTFIXA
INFIXA a / ( b * ( c - ( d + e ) ) ) -> POSTFIXA
INFIXA a1 + ( b2 - ( c3 * ( d4 / e5 ) ) ) -> POSTFIXA
INFIXA a1 + ( b2 - ( c3 * d4 / e5 ) ) -> POSTFIXA
INFIXA a1 + ( b2 - c3 * d4 / e5 ) -> POSTFIXA
INFIXA a1 + b2 - c3 * d4 / e5 -> POSTFIXA
INFIXA a1 + ( b2 - c3 ) * d4 / e5 -> POSTFIXA
INFIXA ( a1 + ( b2 - c3 ) ) * d4 / e5 -> POSTFIXA
INFIXA ( a1 + ( b2 - c3 ) ) * ( d4 / e5 ) -> POSTFIXA
INFIXA ( c and d ) and ( a and b ) -> POSTFIXA
INFIXA c and d and a and b -> POSTFIXA
INFIXA not ( c and d ) and not ( a and b ) -> POSTFIXA
INFIXA not not ( c and d ) and not not ( a and b ) -> POSTFIXA
INFIXA not c and not d and not a and not b -> POSTFIXA
INFIXA not not c and not not d and not not a and not not b -> POSTFIXA
