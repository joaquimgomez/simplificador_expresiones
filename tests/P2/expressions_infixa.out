22
18
(a and (b or (not a)))
(a and b)
(a and b)
(a and b)
(a and b)
(a or b)
(a or b)
((a or (b and a)) or b)
0
0
-10
12
2
0
(((x + 3) + (2 * y)) - (((2 * y) + 3) + x))
(((((x + 3) + (2 * y)) - (2 * y)) + 3) + x)
((x + 12) + (12 + x))
(x + 12)
(12 + x)
0
(x + 12)
((x + 12) * (12 + x))
0
0
(x + 12)
(12 + x)
1
(x + 12)
0
(((T == c) or d) and e)
((T == (c or d)) and e)
(T == (c or (d and e)))
((T == c) or (d and e))
(aa and (bb or (cc == T)))
(aa and ((bb or cc) == T))
(((aa and bb) or cc) == T)
((aa and bb) or (cc == T))
((((a / b) * c) - d) + e)
((((a / b) * c) - d) + e)
((((a / b) * c) - d) + e)
((((a / b) * c) - d) + e)
(((a / b) * c) - (d + e))
((a / b) * (c - (d + e)))
(a / (b * (c - (d + e))))
(a1 + (b2 - (c3 * (d4 / e5))))
(a1 + (b2 - ((c3 * d4) / e5)))
(a1 + (b2 - ((c3 * d4) / e5)))
((a1 + b2) - ((c3 * d4) / e5))
(a1 + (((b2 - c3) * d4) / e5))
(((a1 + (b2 - c3)) * d4) / e5)
((a1 + (b2 - c3)) * (d4 / e5))
((c and d) and (a and b))
(((c and d) and a) and b)
((not (c and d)) and (not (a and b)))
((c and d) and (a and b))
((((not c) and (not d)) and (not a)) and (not b))
(((c and d) and a) and b)
22 
18 
and a or b not a 
and a b 
and a b 
and a b 
and a b 
or a b 
or a b 
or or a and b a b 
0 
0 
-10 
12 
2 
0 
- + + x 3 * 2 y + + * 2 y 3 x 
+ + - + + x 3 * 2 y * 2 y 3 x 
+ + x 12 + 12 x 
+ x 12 
+ 12 x 
0 
+ x 12 
* + x 12 + 12 x 
0 
0 
+ x 12 
+ 12 x 
1 
+ x 12 
0 
and or == T c d e 
and == T or c d e 
== T or c and d e 
or == T c and d e 
and aa or bb == cc T 
and aa == or bb cc T 
== or and aa bb cc T 
or and aa bb == cc T 
+ - * / a b c d e 
+ - * / a b c d e 
+ - * / a b c d e 
+ - * / a b c d e 
- * / a b c + d e 
* / a b - c + d e 
/ a * b - c + d e 
+ a1 - b2 * c3 / d4 e5 
+ a1 - b2 / * c3 d4 e5 
+ a1 - b2 / * c3 d4 e5 
- + a1 b2 / * c3 d4 e5 
+ a1 / * - b2 c3 d4 e5 
/ * + a1 - b2 c3 d4 e5 
* + a1 - b2 c3 / d4 e5 
and and c d and a b 
and and and c d a b 
and not and c d not and a b 
and and c d and a b 
and and and not c not d not a not b 
and and and c d a b 
22 
18 
a b a not or and 
a b and 
a b and 
a b and 
a b and 
a b or 
a b or 
a b a and or b or 
0 
0 
-10 
12 
2 
0 
x 3 + 2 y * + 2 y * 3 + x + - 
x 3 + 2 y * + 2 y * - 3 + x + 
x 12 + 12 x + + 
x 12 + 
12 x + 
0 
x 12 + 
x 12 + 12 x + * 
0 
0 
x 12 + 
12 x + 
1 
x 12 + 
0 
T c == d or e and 
T c d or == e and 
T c d e and or == 
T c == d e and or 
aa bb cc T == or and 
aa bb cc or T == and 
aa bb and cc or T == 
aa bb and cc T == or 
a b / c * d - e + 
a b / c * d - e + 
a b / c * d - e + 
a b / c * d - e + 
a b / c * d e + - 
a b / c d e + - * 
a b c d e + - * / 
a1 b2 c3 d4 e5 / * - + 
a1 b2 c3 d4 * e5 / - + 
a1 b2 c3 d4 * e5 / - + 
a1 b2 + c3 d4 * e5 / - 
a1 b2 c3 - d4 * e5 / + 
a1 b2 c3 - + d4 * e5 / 
a1 b2 c3 - + d4 e5 / * 
c d and a b and and 
c d and a and b and 
c d and not a b and not and 
c d and a b and and 
c not d not and a not and b not and 
c d and a and b and 
